SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialLatencyTracer.h"
+
+#include "Async/Async.h"
+#include "Engine/World.h"
+#include "EngineClasses/SpatialGameInstance.h"
+#include "Interop/Connection/OutgoingMessages.h"
+#include "Utils/SchemaUtils.h"
+
+#include <sstream>
+
+DEFINE_LOG_CATEGORY(LogSpatialLatencyTracing);
+
+namespace
+{
+	// Stream for piping trace lib output to UE output
+	class UEStream : public std::stringbuf
+	{
+		int sync() override
+		{
+			UE_LOG(LogSpatialLatencyTracing, Verbose, TEXT("%s"), *FString(str().c_str()));
+			str("");
+			return std::stringbuf::sync();
+		}
+
+	public:
+		virtual ~UEStream() override
```

Comment:
nit - I think it's more common to not add override (or the redundant virtual) on destructors but it doesn't really matter
even nittier - could do rule of 5 but I don't think we're doing it elsewhere
(feel free to just hit resolve if you disagree)

Reply:
We follow unreal coding standards here, where both are declared - https://docs.unrealengine.com/en-US/Programming/Development/CodingStandard/index.html

And likewise implementing rule of 5, like you say it isn't something we do elsewhere, and this objects usage is well contained.